sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py

```diff
@@ -49,7 +50,7 @@ def create_role_assignment(self, role_scope, role_assignment_name, role_definiti
         assignment = self._client.role_assignments.create(
             vault_base_url=self._vault_url,
             scope=role_scope,
-            role_assignment_name=role_assignment_name,
+            role_assignment_name=kwargs.pop("role_assignment_name", uuid4()),
```

Comment:
This calls `uuid4()` even when `kwargs["role_assignment_name"]` has a value.
```suggestion
            role_assignment_name=kwargs.pop("role_assignment_name", None) or uuid4(),
```

Reply:
Truly absentminded today, missing this and the async client/tests. Thanks for catching it!